src/elements/pages/dev/game-overview.ts

```diff
@@ -0,0 +1,296 @@
+
+	constructor() {
+		super();
+
+		this.validateNamespaceDebounce = new Debounce({
+			delay: timing.milliseconds(500),
+			cb: async () => {
+				let displayName = this.namespaceDisplayNameValue ?? '';
+				let nameId = this.namespaceNameIdValue.length
+					? this.namespaceNameIdValue
+					: utils.convertStringToId(displayName);
+
+				return await global.cloud.validateGameNamespace({
+					gameId: this.game.gameId,
+					nameId,
+					displayName
+				});
+			},
+			completeCb: res => {
+				// Save errors
+				this.validationErrors.load(res.errors.map(err => err.path));
+				this.namespaceIsValid = this.validationErrors.isEmpty();
+				this.loadError = null;
+
+				// Refresh UI
+				this.requestUpdate('validationErrors');
+			}
+		});
+
+		this.validateNamespaceDebounce.onError(async err => {
+			this.loadError = err;
+			this.namespaceIsValid = false;
+		});
+	}
+
+	updated(changedProperties: PropertyValues) {
+		super.updated(changedProperties);
+
+		if (changedProperties.has('game')) {
+			this.fetchData();
+			this.requestUpdate();
+		}
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this.gameStream) this.gameStream.cancel();
+	}
+
+	async fetchData() {
+		if (this.gameStream) {
+			this.gameStream.cancel();
+			this.gameStream = null;
+		}
+
+		try {
+			this.gameStream = await CloudGameCache.watch(this.gameId, async res => {
+				let gameData = res.game;
+				if (gameData) {
+					this.game = gameData;
+					this.requestUpdate('game');
+					return;
+				}
+			});
+		} catch (err) {
+			logging.error('failed to fetch game data', err);
+		}
+	}
+
+	openNamespaceModal() {
+		if (this.game.versions.length == 0) {
+			showAlert(
+				'Cannot create namespace',
+				html`You cannot create a namespace before creating a version first.`,
+				[
+					{
+						label: 'Create A Version',
+						cb: () =>
+							UIRouter.shared.navigate(routes.devVersionDraft.build({ gameId: this.gameId }))
+					},
+					{
+						label: 'Dismiss'
+					}
+				]
+			);
+		} else {
+			this.namespaceModalActive = true;
+		}
+	}
+
+	namespaceModalClose() {
+		this.namespaceModalActive = false;
+	}
+
+	namespaceDisplayNameInput(event: InputUpdateEvent) {
+		this.namespaceDisplayNameValue = event.value;
+
+		this.validateNamespaceDebounce.trigger();
+	}
+
+	namespaceNameIdInput(event: InputUpdateEvent) {
+		this.namespaceNameIdValue = event.value;
+
+		this.validateNamespaceDebounce.trigger();
+	}
+
+	renderCreateNamespaceModal() {
+		let displayName = this.namespaceDisplayNameValue;
+
+		let displayNameErrors = this.validationErrors.findFormatted('display-name');
+		let nameIdErrors = this.validationErrors.findFormatted('name-id');
+
+		return html` <drop-down-modal
+			id="create-namespace-modal"
+			?active=${this.namespaceModalActive}
+			@close=${this.namespaceModalClose.bind(this)}
+		>
+			<modal-body slot="body">
+				<h1>Create your new namespace</h1>
+				<div class="input-group">
+					<h2>Namespace Name</h2>
```

Comment:
User already knows they're making a namespace. Can just call this "Display Name".